Add HassStopMoving intent for covers and valves

ezra-freedman opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR adds the HassStopPosition intent to the intent component, enabling LLM-based assistants (like the MCP server integration) and conversation agents to stop covers and valves mid-operation.

Previously, covers and valves could be positioned via HassSetPosition, but lacked a stop intent for halting mid-operation. This expands the set of control intents by adding the stop functionality, which is particularly useful for:
  - Stopping garage doors mid-operation
  - Halting blinds or shades at a desired position without setting an explicit position value
  - Stopping valves mid-operation
  - Emergency stops for any motorized cover or valve

The intent follows the same pattern as HassSetPosition, implemented as a general intent using DynamicServiceIntentHandler that works across multiple domains (covers and valves) rather than being domain-specific. This approach is preferred over the deprecated HassOpenCover/HassCloseCover pattern which has been replaced by generic turn on/off functionality.

The implementation:
  - Uses DynamicServiceIntentHandler to route to cover.stop_cover or valve.stop_valve based on domain
  - Is automatically exposed to LLM-based conversation agents
  - Works with MCP server and other programmatic intent callers
  - Includes tests covering covers, valves, error handling, and LLM API exposure

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

ezra-freedman committed:
add stop cover intent

ezra-freedman committed:
Add tests for stop cover intent

Tests verify that the HassStopCover intent correctly handles stopping covers by name and device class.

abmantis reviewed (commented):
Why not implement it in the same way as SetPosition, so that it works also for valves?

ezra-freedman committed:
Add HassStop intent for covers and valves

Implements a general stop intent following the HassSetPosition pattern.
Works for both cover and valve domains using DynamicServiceIntentHandler.

ezra-freedman committed:
Merge branch 'dev' into add_intent_stop_cover

Diegorro98 commented on homeassistant/components/intent/__init__.py:
```diff
@@ -433,6 +436,31 @@ def get_domain_and_service(
+            return (COVER_DOMAIN, SERVICE_STOP_COVER)
+
+        if state.domain == VALVE_DOMAIN:
+            return (VALVE_DOMAIN, SERVICE_STOP_VALVE)
```
True, having this in mind, the way it is is already right

abmantis reviewed (approved):
LGTM, but needs a review from one of the codeowners too.

frenck commented:
> > LGTM, but needs a review from one of the codeowners too.
> 
> Is that something I need to request or is it automated?

That is mostly a matter of patience 🙏 

../Frenck  

<p>
  <a href="https://www.linkedin.com/in/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/linkedin.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://youtube.com/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/youtube.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://bsky.app/profile/frenck.social"><img src="https://github.com/frenck/frenck/raw/main/images/bluesky.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://fosstodon.org/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/mastodon.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.instagram.com/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/instagram.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.threads.net/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/threads.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.facebook.com/frenck.dev/"><img src="https://github.com/frenck/frenck/raw/main/images/facebook.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://x.com/frenck"><img src="https://github.com/frenck/frenck/raw/main/images/x.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.tiktok.com/@frenck.nl"><img src="https://github.com/frenck/frenck/raw/main/images/tiktok.svg" width="18" height="18"></a>
</p>

Blogging my personal ramblings at <a href="https://frenck.dev">frenck.dev</a>

arturpragacz reviewed (changes requested):
`HassStop` as a name is too broad, a lot of things can be stopped in principle, but this intent is meant just for covers and valves.

ezra-freedman committed:
Merge branch 'home-assistant:dev' into add_intent_stop_cover

ezra-freedman committed:
Use HassStopPosition as intent name for consistency with HassSetPosition

ezra-freedman committed:
naming consistency

arturpragacz reviewed (changes requested):
I'm not quite sure about the name yet, I'll try to discuss it some more internally.

But first let's improve the PR a bit.

ezra-freedman committed:
Merge branch 'home-assistant:dev' into add_intent_stop_cover

ezra-freedman committed:
PR feedback, moving tests

ezra-freedman commented on tests/components/cover/test_intent.py:
```diff
@@ -83,6 +84,39 @@ async def test_close_cover_intent(hass: HomeAssistant, slots: dict[str, Any]) ->
+    ],
+)
+async def test_stop_cover_position(hass: HomeAssistant, slots: dict[str, Any]) -> None:
+    """Test HassStopPosition intent for covers."""
```
done

ezra-freedman commented on tests/components/valve/test_intent.py:
```diff
@@ -81,3 +82,24 @@ async def test_set_valve_position(hass: HomeAssistant) -> None:
+
+
+async def test_stop_valve_position(hass: HomeAssistant) -> None:
+    """Test HassStopPosition intent for valves."""
```
done

arturpragacz committed:
Update tests/components/intent/test_init.py

ezra-freedman committed:
Merge branch 'dev' into add_intent_stop_cover

ezra-freedman committed:
PR feedback, intent should be called HassStopMoving
